src/business.ts

```diff
@@ -80,38 +95,53 @@ export class BusinessApi {
     error?: MarqetaError,
   }> {
     const searchOptions = snakecaseKeys(search)
-    const resp = await this.client.fire(
-      'GET',
+    const resp = await this.client.fire('GET',
       'businesses',
       { ...searchOptions }
     )
-    // catch 404s...
-    if (resp?.payload?.errorCode >='404000' && resp?.payload?.errorCode <= 404999) {
-      const error = resp?.payload?.errorMessage
+    // catch any errors...
+    if (resp?.payload?.errorCode) {
       return {
         success: false,
         error: {
           type: 'marqeta',
-          error,
+          error: resp?.payload?.errorMessage,
           status: resp?.payload?.errorCode,
-          marqetaStatus: resp?.payload?.errorCode
         },
       }
     }
-    // catch all other errors...
-    if (resp?.payload?.errorCode > 404999){
-      const error = resp?.payload?.errorMessage
+    return { success: !resp?.payload?.errorCode, body: { ...resp.payload } }
+  }
+
+  /*
+   * Function to take the attributes of a new Business account, create that
+   * in Marqeta, and return the account information. If no attributes are
+   * provided, a new Business account will still be created and its
+   * token ID returned.
+   */
+  async create(create: {
+     business: Business
+  }): Promise<{
+    success: boolean,
+    body?: Business,
+    error?: MarqetaError,
+  }> {
+    const searchOptions = snakecaseKeys(create?.business)
```

Comment:
The way you have the function signature set up, `create` is not optional, and it kinda messes up the flow... I mean to create something like this you'd have to call it with:

```typescript
const resp = await client.business.create({ business: {
  ...
}})
```

why not just drop the nesting and go with:

```typescript
  async create(business: Business): Promise<{
     success: boolean,
     body?: Business,
     error?: MarqetaError,
   }> {
```

so that `business` is the "map of data" that holds everything?

-----

and why is this called:

```typescript
    const searchOptions = snakecaseKeys(create?.business)
```

`searchOptions`? This would really need to be:

```typescript
    const searchOptions = snakecaseKeys(business)
```

anyway, but the name just doesn't make sense to me... What about dropping this line altogether and using:

```typescript
     const resp = await this.client.fire('POST',
       'businesses',
       {},
       snakecaseKeys(business))
     // catch any errors...
     if (resp?.payload?.errorCode) {
```

then you don't need it, and the code reads a lot cleaner - Right?

Reply:
agreed.